iothub/service/src/DirectMethod/DirectMethodRequestOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Optional parameters to execute a direct method with.
+    /// </summary>
+    public class DirectMethodRequestOptions
+    {
+        /// <summary>
+        /// The timeout (in seconds) before the direct method request will fail if the device fails to respond to the request.
+        /// This timeout may happen if the target device is slow in handling the direct method.
+        /// </summary>
+        public int? ResponseTimeout { get; set; }
```

Comment:
Note that we used to accept this as a timespan. I'm in favor of using int? here instead because the service has pretty strict limits on what this value can be (5-300). On top of that, exposing these as a timespan might falsely convince users that there is a level of granularity beyond seconds. Finally, if users try to pass in something like

```csharp
options.ResponseTimeout = TimeSpan.FromMilliseconds(50);
```

Then the SDK/service will end up interpreting that as 0 seconds and throw an exception that would confuse users.

The typical code for a user would go from

```csharp
options.ResponseTimeout = TimeSpan.FromSeconds(50);
```

to

```csharp
options.ResponseTimeout = 50;
```